packages/core/phase2/lib/getOperationSequence/areAllResultsAlreadyPresentOnPnc/isMatchToPncAdjAndDis/isMatchToPncAdjAndDis.ts

```diff
@@ -1,57 +1,52 @@
-  offenceIndex: number,
-  offenceReasonSequence: string | undefined
-): ExceptionResult<boolean> => {
+const findPncCourtCase = (aho: AnnotatedHearingOutcome, offence: Offence) => {
   const courtCaseReference =
-    courtCaseReferenceNumber ?? aho.AnnotatedHearingOutcome.HearingOutcome.Case.CourtCaseReferenceNumber
+    offence.CourtCaseReferenceNumber ?? aho.AnnotatedHearingOutcome.HearingOutcome.Case.CourtCaseReferenceNumber
 
-  if (!courtCaseReference) {
-    return { value: false, exceptions: [] }
-  }
-
-  const courtCase = aho.PncQuery?.courtCases?.find((x) => x.courtCaseReference === courtCaseReference)
+  return courtCaseReference
+    ? aho.PncQuery?.courtCases?.find((x) => x.courtCaseReference === courtCaseReference)
+    : undefined
+}
 
-  if (!courtCase) {
+const isMatchToPncAdjAndDis = (
+  aho: AnnotatedHearingOutcome,
+  offence: Offence,
+  offenceIndex: number
+): ExceptionResult<boolean> => {
+  if (!offence.Result || !isNonEmptyArray(offence.Result)) {
     return { value: false, exceptions: [] }
   }
 
-  const pncOffences = courtCase.offences
-
-  if (!pncOffences || pncOffences.length == 0) {
+  const pncCourtCase = findPncCourtCase(aho, offence)
+  if (!pncCourtCase || pncCourtCase.offences.length === 0) {
     return { value: false, exceptions: [] }
   }
 
-  const adjFromAho = getAdjFromAho(results, aho.AnnotatedHearingOutcome.HearingOutcome.Hearing.DateOfHearing)
-
-  const matchingPncAdjudications = pncOffences.filter((pncOffence) =>
-    isMatchToPncAdj(adjFromAho, pncOffence, offenceReasonSequence)
+  const offenceReasonSequence = offence.CriminalProsecutionReference?.OffenceReasonSequence ?? undefined
+  const adjFromAho = createPncAdjudicationByAho(
+    offence.Result,
+    aho.AnnotatedHearingOutcome.HearingOutcome.Hearing.DateOfHearing
   )
-
-  if (!matchingPncAdjudications || matchingPncAdjudications.length == 0) {
-    return { value: false, exceptions: [] }
-  }
-
   const exceptions: Exception[] = []
-  const isThereMatchingPncAdjudications = matchingPncAdjudications.some((pncAdj) => {
-    const { value: allPncDisposalsMatch, exceptions: allPncDisposalsMatchExceptions } =
-      allRecordableResultsMatchAPncDisposal(results, pncAdj.disposals ?? [], aho, offenceIndex)
-    exceptions.push(...allPncDisposalsMatchExceptions)
+  const isMatchToPncAdjudicationAndDisposals = pncCourtCase.offences.some((pncOffence) => {
+    if (!isMatchToPncAdjudication(adjFromAho, pncOffence, offenceReasonSequence)) {
+      return false
+    }
+
+    const pncDisposalsMatchResult = areResultsMatchAPncDisposal(offence, offenceIndex, pncOffence.disposals ?? [])
+    exceptions.push(...pncDisposalsMatchResult.exceptions)
 
-    return allPncDisposalsMatch
+    return pncDisposalsMatchResult.value
   })
 
-  return { value: isThereMatchingPncAdjudications, exceptions }
+  return { value: isMatchToPncAdjudicationAndDisposals, exceptions }
 }
 
 export default isMatchToPncAdjAndDis
```

Comment:
Could we name this `isMatchToPncAdjudicationAndDisposals`?

Reply:
Renamed. ✔️